contracts/src/instances/eeth/EETHBase.sol

```diff
@@ -0,0 +1,160 @@
+            msg.sender,
+            address(this),
+            baseAmount
+        );
+        if (!result) {
+            revert IHyperdrive.TransferFailed();
+        }
+    }
+
+    /// @dev Process a withdrawal in base and send the proceeds to the
+    ///      destination.
+    function _withdrawWithBase(
+        uint256, // unused
+        address, // unused
+        bytes calldata // unused
+    ) internal pure override returns (uint256) {
+        // eETH withdrawals aren't necessarily instantaneous. Users that want
+        // to withdraw can manage their withdrawal separately.
+        revert IHyperdrive.UnsupportedToken();
+    }
+
+    /// @dev Process a withdrawal in vault shares and send the proceeds to the
+    ///      destination.
+    /// @param _shareAmount The amount of vault shares to withdraw.
+    /// @param _destination The destination of the withdrawal.
+    function _withdrawWithShares(
+        uint256 _shareAmount,
+        address _destination,
+        bytes calldata // unused
+    ) internal override {
+        // Convert the vault shares to base.
+        uint256 baseAmount = _convertToBase(_shareAmount);
+
+        // NOTE: The eETH transfer function converts from base to shares under
+        // the hood using `sharesForAmount(_amount)`.
+        // Transfer the stETH shares to the destination.
+        bool result = IeETH(address(_vaultSharesToken)).transfer(
+            _destination,
+            baseAmount
+        );
+        if (!result) {
+            revert IHyperdrive.TransferFailed();
+        }
+    }
+
+    /// @dev Convert an amount of vault shares to an amount of base.
+    /// @param _shareAmount The vault shares amount.
+    /// @return The base amount.
+    function _convertToBase(
+        uint256 _shareAmount
+    ) internal view override returns (uint256) {
+        return
+            EETHConversions.convertToBase(
+                liquidityPool,
+                _vaultSharesToken,
+                _shareAmount
+            );
+    }
+
+    /// @dev Convert an amount of base to an amount of vault shares.
+    /// @param _baseAmount The base amount.
+    /// @return The vault shares amount.
+    function _convertToShares(
+        uint256 _baseAmount
+    ) internal view override returns (uint256) {
+        return
+            EETHConversions.convertToShares(
+                liquidityPool,
+                _vaultSharesToken,
+                _baseAmount
+            );
+    }
+
+    /// @dev Gets the total amount of shares held by the pool in the yield
+    ///      source.
+    /// @return shareAmount The total amount of shares.
+    function _totalShares()
+        internal
+        view
+        override
+        returns (uint256 shareAmount)
+    {
+        return _vaultSharesToken.balanceOf(address(this));
+    }
+
+    /// @dev We override the message value check since this integration is
+    ///      not payable.
+    function _checkMessageValue() internal pure override {}
```

Comment:
Confused as to why we're checking `msg.value` in `_depositWithBase` if this integration isn't payable

Reply:
oh. the comment needs to be updated